core/src/main/java/com/backblaze/b2/client/structures/B2UploadProgress.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017, Backblaze Inc. All Rights Reserved.
+ * License https://www.backblaze.com/using_b2_code.html
+ */
+package com.backblaze.b2.client.structures;
+
+/**
+ * The B2UploadProgress summarizes the status of an upload.
```

Comment:
It's not clear to me how to interpret this structure.  Maybe I haven't read enough about the other classes?

Is this the progress for one part?  Does the state diagram in the enum class apply to each part?